app/assets/javascripts/to_do_lists.coffee

```diff
@@ -1,3 +1,62 @@
 # Place all the behaviors and hooks related to the matching controller here.
 # All this logic will automatically be available in application.js.
 # You can use CoffeeScript in this file: http://coffeescript.org/
+
+# permit functions for eval
+permit_functions = ['expand_on_click', 'collapse_on_click']
+
+# hide element
+hide_element = (element) ->
+  element.addClass('hidden')
+
+# show element
+show_element = (element) ->
+  element.removeClass('hidden')
+
+#make focus on element
+make_focus_on = (element) ->
+  element.val('').focus()
+
+# expand a form and hide a button
+expand_on_click = (element) ->
+  $(element).click (event) ->
+    event.preventDefault()
+    hide_element $(this)
+    show_element $(this).parent().find('form')
+    make_focus_on $(this).parent().find("input[type='text']")
```

Comment:
@vlasiak, Note that calling `$(this).parent()` twice, you make a query to the dom via jQuery two times. I suggest to extract such queries into local variables. Please name them using leading `$` in order to clearly understand that it is a jQuery object, for example:
```javascript
$container = $(this).parent()
$container.find('form')
$container.find("input[type='text']")
```